test/parallel/test-cluster-primary-error.js

```diff
@@ -87,13 +99,12 @@ if (cluster.isWorker) {
   primary.on('exit', common.mustCall((code) => {
     // Check that the cluster died accidentally (non-zero exit code)
     assert.strictEqual(code, 1);
-
-    // XXX(addaleax): The fact that this uses raw PIDs makes the test inherently
-    // flaky – another process might end up being started right after the
-    // workers finished and receive the same PID.
     const pollWorkers = () => {
       // When primary is dead all workers should be dead too
-      if (workers.some((pid) => common.isAlive(pid))) {
+      if (workers.some((pid) => {
+        const filepath = `${tmpdir.path}/${pid}-test-cluster-primary-error`;
+        return common.isAlive(pid) && fs.existsSync(filepath);
+      })) {
```

Comment:
It's still using raw PIDs, isn't it? Shouldn't the TODO comment stays?

Reply:
Yes, I add it back :-).